.github/workflows/upload-build-to-s3.yaml

```diff
@@ -62,48 +66,220 @@ jobs:
+      - name: configure aws credentials
+        uses: aws-actions/configure-aws-credentials@v2
         with:
-          name: finch.macos-x86_64
-          path: finch.*.x86_64.tar.gz
-          if-no-files-found: error
+          role-to-assume: ${{ secrets.ROLE }}
+          role-session-name: dependency-upload-session
+          aws-region: ${{ secrets.REGION }}
+
+      - name: generate pkg
+        run: |
+          ./installer-builder/tools/release-installer.sh x86_64 ${GITHUB_REF_NAME} ${{ secrets.INSTALLER_PRIVATE_BUCKET_NAME }} ${{ secrets.EXECUTABLE_BUCKET }} ${{ secrets.PKG_BUCKET }} ${{ secrets.NOTARIZATION_ACCOUNT }} ${{ secrets.NOTARIZATION_CREDENTIAL }}
+        shell: zsh {0}
 
-  release:
-    runs-on: ubuntu-latest
-    timeout-minutes: 5
+  macos-aarch64-pkg-test:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [
+              [self-hosted, macos, arm64, 11, release], 
+              [self-hosted, macos, arm64, 12, release], 
+              [self-hosted, macos, arm64, 13, release]
+            ]
+    runs-on: ${{ matrix.os }}
     needs:
-      - macos-x86_64-build
-      - macos-aarch64-build
+      - macos-aarch64-pkg-build
+    timeout-minutes: 60
+    env:
+      ACCESS_TOKEN: ${{ secrets.FINCH_BOT_TOKEN }}
     steps:
-      - uses: actions/checkout@v3
+      - name: Checkout the tag
+        uses: actions/checkout@v3
         with:
+          ref: ${{ github.ref_name }}
           fetch-depth: 0
           persist-credentials: false
-
+          submodules: true
+      - uses: actions/setup-go@v4
+        with:
+          go-version-file: go.mod
+          cache: true
+      - name: Clean up previous files
+        run: |
+          sudo rm -rf /opt/finch
+          sudo rm -rf ~/.finch
+          sudo rm -rf ./_output
+          if pgrep '^qemu-system'; then
+            sudo pkill '^qemu-system'
+          fi
+          if pgrep '^socket_vmnet'; then
+            sudo pkill '^socket_vmnet'
+          fi
       - name: configure aws credentials
         uses: aws-actions/configure-aws-credentials@v2
         with:
           role-to-assume: ${{ secrets.ROLE }}
-          role-session-name: dependency-upload-session
+          role-session-name: download-installer-session
           aws-region: ${{ secrets.REGION }}
+      - name: Download from S3
+        run: |
+          aws s3 cp s3://${{ secrets.INSTALLER_PRIVATE_BUCKET_NAME }}/Finch-${GITHUB_REF_NAME}-aarch64.pkg Finch-${GITHUB_REF_NAME}-aarch64.pkg
+        shell: zsh {0}
+      - name: Silently install
+        run: sudo installer -pkg Finch-${GITHUB_REF_NAME}-aarch64.pkg -target /
+      - name: Install Rosetta 2
+        run: echo "A" | softwareupdate --install-rosetta || true 
+      - name: Build project
+        run: |
+          brew install go lz4 automake autoconf libtool
+          export PATH="/opt/homebrew/opt/libtool/libexec/gnubin:$PATH"
+          make
+        shell: zsh {0}
+      - name: Multiple instances of Finch test
```

Comment:
Just curious about why we test with 2 instances of finch?

Reply:
You can build Finch from source locally and have Finch installed to Application at the same time. This will result in potential issue with override the persistent disk file and conflict. The persistent disk has the file name related with the Finch path, so the 2 instances will have different files. This test will make sure local Finch and /Application Finch both work and don't have conflict.